crates/cli/cli/src/commands/connection/create.rs

```diff
@@ -0,0 +1,105 @@
+use std::time::Duration;
+
+use hermes_cli_framework::command::Runnable;
+use hermes_cosmos_client_components::types::connection::CosmosInitConnectionOptions;
+use hermes_cosmos_relayer::contexts::builder::CosmosBuilder;
+use hermes_relayer_components::build::components::relay::build_from_chain::BuildRelayFromChains;
+use hermes_relayer_components::build::traits::components::relay_builder::RelayBuilder;
+use hermes_relayer_components::build::traits::target::relay::RelayAToBTarget;
+use hermes_relayer_components::logger::traits::log::CanLog;
+use hermes_relayer_components::relay::impls::connection::bootstrap::CanBootstrapConnection;
+use hermes_relayer_runtime::types::log::level::LogLevel;
+use ibc_relayer_types::core::ics03_connection::version::Version;
+use ibc_relayer_types::core::ics24_host::identifier::{ChainId, ClientId};
+use oneline_eyre::eyre::eyre;
+
+use crate::Result;
+
+#[derive(Debug, clap::Parser)]
+pub struct ConnectionCreate {
+    /// Identifier of chain A
+    #[clap(
+        long = "chain-a",
+        required = true,
+        value_name = "CHAIN_ID_A",
+        help_heading = "REQUIRED"
+    )]
+    chain_id_a: ChainId,
+
+    /// Identifier of client A
+    #[clap(
+        long = "client-a",
+        required = true,
+        value_name = "CLIENT_ID_A",
+        help_heading = "REQUIRED"
+    )]
+    client_id_a: ClientId,
+
+    /// Identifier of chain B
+    #[clap(
+        long = "chain-b",
+        required = true,
+        value_name = "CHAIN_ID_B",
+        help_heading = "REQUIRED"
+    )]
+    chain_id_b: ChainId,
+
+    /// Identifier of client B
+    #[clap(
+        long = "client-b",
+        required = true,
+        value_name = "CLIENT_ID_B",
+        help_heading = "REQUIRED"
+    )]
+    client_id_b: ClientId,
+}
+
+impl Runnable for ConnectionCreate {
```

Comment:
In the future, we can make the command runner context-generic by implementing as something like:

```rust
impl<Build, Relay, ChainA, ChainB> CommandRunner<Build> for ConnectionCreate
where
    Build: CanBuildRelay<RelayAToBTarget>,
    Build::BiRelay: HasTwoWayRelayTypes<ChainA = ChainA, ChainB = ChainB, RelayAToB = Relay>,
    ChainA: HasIbcChainTypes<ChainB, ChainId = ChainId, ClientId = ClientId>
        + HasInitConnectionOptionsType<ChainB, InitConnectionOptions = CosmosInitConnectionOptions>,
    ChainB: HasIbcChainTypes<ChainA, ChainId = ChainId, ClientId = ClientId>,
    Relay: HasRelayChains<SrcChain = ChainA, DstChain = ChainB> + CanBootstrapConnection,
    Build::Error: Debug,
    Relay::Error: Debug,
{
    async fn run(&self, build: &Build) -> Result<()> {
        let relay = build
            .build_relay(
                RelayAToBTarget,
                &self.chain_id_a,
                &self.chain_id_b,
                &self.client_id_a,
                &self.client_id_b,
            )
            .await
            .map_err(|e| eyre!("Failed to build relay: {:?}", e))?;

        let options = CosmosInitConnectionOptions {
            delay_period: Duration::from_secs(0),
            connection_version: Version::default(),
        };

        let (connection_id_a, connection_id_b) =
            relay.bootstrap_connection(&options).await.map_err(|e| {
                eyre!(
                    "Failed to create connection: connection handshake failed: {:?}",
                    e
                )
            })?;

        Ok(())
    }
}
```

While refactoring, it is still possible to define context-specific implementations such as `impl CommandRunner<CosmosBuilder> for ConnectionCreate { ... }`